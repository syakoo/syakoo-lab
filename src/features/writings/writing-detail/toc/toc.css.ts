import { style } from "@vanilla-extract/css";
import { recipe } from "@vanilla-extract/recipes";

import { theme } from "@/shared/design-system/theme.css";

export const tocStyles = {
  list: style({
    marginTop: theme.space[50],
    maxWidth: theme.size[50],
    display: "flex",
    flexDirection: "column",
    gap: theme.space[25],
  }),
  item: recipe({
    base: {
      padding: theme.space[25],
      color: theme.color.text.secondary,
      fontSize: theme.fontSize[75],
    },

    variants: {
      active: {
        true: {
          color: theme.color.text.primary,
        },
      },
    },
  }),
};
