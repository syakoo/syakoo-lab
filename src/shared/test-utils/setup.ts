import { exampleMDXComponent } from "@/features/mdx/mocks/fixture-mdxcomponent";

import { loadEnv } from "./nextjs/env";

// NOTE: `@mdx-js/mdx` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
jest.mock("@mdx-js/mdx", () => {
  return {
    runSync: jest
      .fn()
      .mockImplementation(() => ({ default: exampleMDXComponent })),
  };
});

// NOTE: `mermaid` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
jest.mock("mermaid", () => ({
  initialize: () => {
    return;
  },
  run: () => {
    return;
  },
}));

loadEnv();

jest.spyOn(Math, "random").mockImplementation(() => 0.5866245602626035);
