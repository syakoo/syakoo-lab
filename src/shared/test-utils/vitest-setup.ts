import { vi } from "vitest";

import { exampleMDXComponent } from "@/features/mdx/mocks/fixture-mdxcomponent";

import { loadEnv } from "./nextjs/env";

// NOTE: Next.jsのImageコンポーネントをテスト用にモック
vi.mock("next/image", () => ({
  default: "img",
}));

// NOTE: `@mdx-js/mdx` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
vi.mock("@mdx-js/mdx", () => {
  return {
    runSync: vi
      .fn()
      .mockImplementation(() => ({ default: exampleMDXComponent })),
  };
});

// NOTE: `mermaid` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
vi.mock("mermaid", () => ({
  default: {
    initialize: () => {
      return;
    },
    run: () => {
      return;
    },
  },
  initialize: () => {
    return;
  },
  run: () => {
    return;
  },
}));

loadEnv();

vi.spyOn(Math, "random").mockImplementation(() => 0.5866245602626035);
