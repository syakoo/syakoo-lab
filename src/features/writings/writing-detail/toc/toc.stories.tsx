import type { Meta } from "@storybook/nextjs";

import { TOC } from ".";
import { TOCView } from "./toc";

const meta = {
  component: TOC,
  parameters: {
    layout: "fullscreen",
  },
} satisfies Meta<typeof TOC>;

export default meta;

export const Sample = () => {
  return (
    <TOCView
      activeLabel="基本的な概念"
      items={[
        { label: "導入", href: "#" },
        { label: "基本的な概念", href: "#" },
        { label: "開発環境のセットアップ", href: "#" },
        { label: "プログラミングの基礎", href: "#" },
      ]}
    />
  );
};
