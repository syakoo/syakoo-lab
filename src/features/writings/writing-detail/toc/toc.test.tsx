import { testStories } from "@/shared/test-utils/test-stories";

import * as stories from "./toc.stories";

describe("TOC", () => {
  testStories(stories);
});
