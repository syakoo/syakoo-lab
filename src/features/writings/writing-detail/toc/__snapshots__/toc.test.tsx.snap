// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TOC スナップショット Sample 1`] = `
<body>
  <div>
    <nav>
      <p
        class="text_textStyle__1djq151e text_textStyle_color_currentColor__1djq151i text_textStyle_variants_color_currentColor__1djq1513 text_textStyle_size_100__1djq151l text_textStyle_variants_size_100__1djq1516 text_textStyle_weight_bold__1djq151t"
      >
        目次
      </p>
      <ul
        class="toc_tocStyles_list__1yz9ns40"
      >
        <li
          class="toc_tocStyles_item__1yz9ns41"
        >
          <a
            class="link_linkStyle__1vz8ftz0 link_linkStyle_noHovered_true__1vz8ftz3 link_linkStyle_display_block__1vz8ftz4"
            href="#"
          >
            導入
          </a>
        </li>
        <li
          class="toc_tocStyles_item__1yz9ns41 toc_tocStyles_item_active_true__1yz9ns42"
        >
          <a
            class="link_linkStyle__1vz8ftz0 link_linkStyle_noHovered_true__1vz8ftz3 link_linkStyle_display_block__1vz8ftz4"
            href="#"
          >
            基本的な概念
          </a>
        </li>
        <li
          class="toc_tocStyles_item__1yz9ns41"
        >
          <a
            class="link_linkStyle__1vz8ftz0 link_linkStyle_noHovered_true__1vz8ftz3 link_linkStyle_display_block__1vz8ftz4"
            href="#"
          >
            開発環境のセットアップ
          </a>
        </li>
        <li
          class="toc_tocStyles_item__1yz9ns41"
        >
          <a
            class="link_linkStyle__1vz8ftz0 link_linkStyle_noHovered_true__1vz8ftz3 link_linkStyle_display_block__1vz8ftz4"
            href="#"
          >
            プログラミングの基礎
          </a>
        </li>
      </ul>
    </nav>
  </div>
</body>
`;
